lib/cfn-model/parser/cfn_parser.rb

```diff
@@ -103,6 +124,22 @@ def transform_hash_into_model_elements(cfn_hash, cfn_model)
     cfn_model
   end
 
+  def transform_hash_into_model_elements_with_numbers(cfn_hash, cfn_model)
+    cfn_hash['Resources'].each do |resource_name, resource|
+      resource_class = class_from_type_name resource['Type']['value']
+
+      resource_object = resource_class.new(cfn_model)
+      resource_object.logical_resource_id = { name: resource_name, line: resource['Type']['line'] }
```

Comment:
i rarely ask for comments lol - but i think it might be good to describe that value/line comes from the ToRubyWithLineNumbers and explain its an annotated version of the YML model.

if i understand everything else... i think i prefer an extra hash in cfn_model for recording line numbers by resource id - that way no change of interface, no impact to the rules (i believe?) and the results processor that looks at violations can do the lookup